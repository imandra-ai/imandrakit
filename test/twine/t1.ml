type foo = {
  a: int;
  b: float;
}
[@@deriving twine, show]

let foo1 = { a = 7; b = 2.0 }
let s = Imandrakit_twine.Encode.to_string foo_to_twine foo1;;

Printf.printf "serialized %s to %S\nhex:\n%s\n" (show_foo foo1) s
  (Hex.hexdump_s (Hex.of_string s))
;;

Printf.printf "len: %d\n" (String.length s)

let foo2 = Imandrakit_twine.Decode.decode_string foo_of_twine s;;

Format.printf "foo after roundtrip: %a@." pp_foo foo2
