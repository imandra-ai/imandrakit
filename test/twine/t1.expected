serialized { T1.a = 7; b = 2. } to "b\0231\000\000\000\000\000\000\000@\n"
hex:
00000000: 6217 3100 0000 0000 0000 400a            b.1.......@.

len: 12
foo after roundtrip: { T1.a = 7; b = 2. }
